### PR TITLE
feat: add custom processor module to add ScaffoldedFrom relation

### PR DESCRIPTION
**Pull request by Zaperex**
### Description
Adds a custom catalog processor that creates relations between catalog entities and the scaffolder templates that created them based on the `spec.scaffoldedFrom` field in the `catalog-info.yaml` of the catalog entities.

image

### What issue(s) does this fix?
Fixes: [RHIDP-2156](https://issues.redhat.com/browse/RHIDP-2156)

## How to test changes / Special notes to the reviewer

1. Install the module using the instructions in the README.md
2. Ingest an entity with a `spec.scaffoldedFrom` field that matches the `entityRef` of a template entity in the catalog (ex: `template:default/test-template`). You can potentially use the following example template: https://github.com/janus-idp/backstage-plugins/blob/main/plugins/scaffolder-annotator-action/examples/templates/01-scaffolder-template.yaml
3. Observe the relation graph of the entity in the catalog (or view the raw YAML/JSON of the entity and find the `scaffoldedFrom` relation in the `relations` array)

### CODE REVIEW

**Comment by Zaperex** [comment]
Used in conjunction with https://github.com/janus-idp/backstage-plugins/pull/1567

File: plugins/catalog-backend-module-scaffolder-relation-processor/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Extension of the `spec` field of the entity model
+ * Used to form relations between entities and the scaffolder templates that generated them
+ *
+ * @public
+ */
+export type ScaffoldedBySpec = {
```
Comment (PatAKnight) [inline plugins/catalog-backend-module-scaffolder-relation-processor/src/types.ts:7]: Change to `ScaffoldedFromSpec` for consistency.

**Comment by Zaperex** [comment]
Hmm that's weird, the template should have the reverse relation in the dependency graph.

**Comment by Zaperex** [comment]
It should work if you manually add it though (as long as the referenced template exists)? That's how I initially tested it before having the action.

**Pull request merged**